### ISSUE TITLE
Functional testing requirements for origami components

### ISSUE THREAD

**Issue by depittman** [Pearson-Higher-Ed/he-origami#12]
We should agree on a standardized approach for functional testing of components hosted in origami, and update the developer guidelines to describe the agreed upon approach. 

Functional testing, for the sake of this conversation, is defined as "in browser" testing of the component using automation (selenium, etc...). By definition this means that the components under test are not "library" components only but also have a visualization aspect.

Initial thoughts for how to approach functional testing:
* Require a demo app for every displayable component, including a mock data route (if required) in a common server language (express, hapi, etc...) 
* Include WebDriverJS based (or another framework?) tests with each displayable component.

**Comment by aarmour** [comment]
[FT](https://github.com/Financial-Times/ft-origami/pull/395) is using [BrowserStack](https://www.browserstack.com/).

**Comment by dlamp** [comment]
I agree we need to do some automated UI testing.  I do not think we need to require a demo app.  I am thinking we need test runner like Karma and have a test pattern to follow that should suffice.

Example Test Pattern:

1. Render
2. Find node
3. Verify
